### PR TITLE
Cleanup SFTP url parsing for StreamingDataset

### PR DESCRIPTION
**Pull request by abhi-mosaic**
This PR allows SFTP urls passed to StreamingDataset to accept the same information (username, port, etc.) that we currently can for our SFTPObjectStore. This also eliminates the need for the `COMPOSER_SFTP_USERNAME` enviornment variable.

### CODE REVIEW

File: composer/datasets/streaming/download.py [hunk 1]
```diff
@@ -66,7 +77,11 @@ def download_from_sftp(remote: str, local: str) -> None:
     with SSHClient() as ssh_client:
         # Connect SSH Client
         ssh_client.set_missing_host_key_policy(AutoAddPolicy)
```
Comment (ravi-mosaicml) [inline composer/datasets/streaming/download.py:79]: Ooof ... I missed this line before, but do not think we should default to this. OK making this an option, but would strongly prefer to default to the RejectPolicy (as does paramiko).

CC @nlsapp @hanlint  for security review

**Review by ravi-mosaicml — APPROVED** [review]
LGTM, though see the (unrelated) comment re: the missing host key policy.

I'll approve this PR for now, though let's make sure to address it.

File: composer/datasets/streaming/download.py (hunk 1 above)
Comment (abhi-mosaic) [reply composer/datasets/streaming/download.py:79]: Ok, is this the same behavior with SFTPObjectStore? Like do we have host keys setup in the docker containers? I can test it.

Comment (ravi-mosaicml) [reply composer/datasets/streaming/download.py:79]: The SFTPObjectStore defaults to reject, but has a config option to override this.

Comment (nlsapp) [reply composer/datasets/streaming/download.py:79]: Should it reject by default and force the user to provide host keys as an argument to mcli (or like, an env var secret/configmap): Yes

Should we have an option, that allows a user to decide if hostkeys are ignored: Also yes

But def not the default and ideally we should warn them: something something "You are ignoring SFTP Host Keys, THIS IS UNSAFE!" or something

Comment (abhi-mosaic) [reply composer/datasets/streaming/download.py:79]: Gonna address this in a followup PR, needs some coordination with the SFTPObjectStore and launch interface.

**Pull request merged**